src/ds/logger.h

```diff
@@ -129,12 +132,21 @@ namespace logger
         j["m"].dump());
     }
 
-    void write(const std::string& log_line) override
+    virtual void write(const std::string& log_line) override
     {
       dump(log_line);
     }
   };
 
+  class JsonConsoleLogger : public JsonLogger
+  {
+  public:
+    void write(const std::string& log_line) override
```

Comment:
I think we just need to set the `ostream f` member to `std::cout`? So rather than overriding write, we could set `f` in a constructor for `JsonConsoleLogger`.